### PR TITLE
use hostip if 0.0.0.0 is returned

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/7600

File: code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/data/metadata/common/ContainerMetaData.java [hunk 1]
```diff
@@ -178,7 +178,11 @@ public void setInstanceAndHostMetadata(Instance instance, HostMetaData hostMetaD
                     if (StringUtils.isEmpty(port.getIpAddress())) {
                         newPorts.add(hostIp + ":" + portObj);
                     } else {
```
Comment (cjellick) [inline code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/data/metadata/common/ContainerMetaData.java]: stupid nitpick: I think an `} else if (port.getIpAddress().equals("0.0.0.0")) {`
would be better.

Another (though slightly less stupid) nitpick: "0.0.0.0".equals(port.getIpAddress()) is safer as it protects against a null pointer when getIPAddress() returns null

Comment (StrongMonkey) [inline code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/data/metadata/common/ContainerMetaData.java]: roger roger!🤔

**Comment by cjellick** [comment]
LGTM

**Pull request merged**